### ISSUE TITLE
Constructors from std::array and std::experimental::dynarray

### ISSUE THREAD

**Issue by insertinterestingnamehere** [libdynd/libdynd#575]
The standard library has a few more array-like types. We should probably support constructing dynd::arrays from them.

**Comment by izaid** [comment]
Yep, we should support std::array. I didn't even know dynarray was a thing, but apparently it's been dropped from C++14?

**Comment by insertinterestingnamehere** [comment]
Yep, it's not in C++14, but it is included as an optional spec for the compilers to implement. Supposedly some of it's behavior had unclear technical details. Libc++ already has a partial implementation though. `std::array` is the big one we'll want to support, but I threw dynarray in too since both constructors aren't likely to take much extra time to implement. It's more a matter of anyone getting to it.
